README.md

```diff
@@ -30,117 +30,44 @@ function TodoListItem (sources) {
-
-Say we have a `TodoList` function, and we want to be able to add new `TodoListItem`s.
+To actually populate the collection, you pass an `add$` stream. Its emitted values may be sources objects, which will be merged with the sources object you passed when you created the `collection$`. This is useful for passing `props$`.
 
 ```js
-function TodoList (sources) {
-  const todoListItems = Collection(TodoListItem, sources);
-
-  const sinks = {
-    DOM: xs.of(
-      div('.todo-list', [
-        button('.add-todo', 'Add todo')
-      ])
-    )
-  }
-
-  return sinks;
-}
+const todoListItems$ = Collection(TodoListItem, sources, xs.of(additionalSources));
 ```
 
-We want to add a new TodoListItem to the collection when we click the 'Add todo' button.
-
-To do this we need to get a stream of reducers to update the collection.
-
-First we declare our reducer.
+`add$` can emit an array, if multiple items should be added at once.
 
 ```js
-function addItemReducer(todoListItems) {
-  return todoListItems.add();
-}
+const todoListItems$ = Collection(TodoListItem, sources, xs.of([firstSources, secondSources]));
 ```
 
-This will return a new copy of todoListItems with a new item added.
-
-Now we can take a stream of click events on the 'Add todo' button.
+`Collection()` returns a stream with arrays of items as values. Those arrays are cloned from internal ones, so changes will not impact the state of the `collection$`.
 
-```js
-const addTodoClick$ = sources.DOM.select('.add-todo').events('click');
-```
-
-We then map over our `addTodoClick` stream to make our `reducer` stream.
-
-```js
-const addTodoReducer$ = addTodoClick$.mapTo(addItemReducer);
-```
-
-We can then build our collection stream by folding over our `reducer` stream.
-
-```js
-const todoListItems$ = addTodoReducer$
-  .fold((items, reducer) => reducer(items), todoListItems);
-```
-
-There are a few things going in this line.
- * We're folding over `addTodoReducer$`, which is a stream of functions that take `todoListItems` and return updated `todoListItems`
- * We're calling each `reducer` and passing the current `items` into it. Each `reducer` returns an updated `items`.
- * We're passing `todoListItems` as the initial value to `fold`, which is the empty collection we made above in our `TodoList` function.
+Collections are **immutable**. This is because in Cycle.js values that change are represented as streams.
 
- If we put it all together in our `TodoList` it looks like this:
+If we put it all together in our `TodoList` it looks like this:
 
 ```js
-function addItemReducer(todoListItems) {
-  return todoListItems.add();
-}
-
 function TodoList (sources) {
-  const todoListItems = Collection(TodoListItem, sources);
-
-  const addTodoClick$ = sources.DOM.select('.add-todo').events('click');
+  const addTodo$ = sources.DOM
+    .select('.add-todo')
+    .events('click')
+    .mapTo(null); // to prevent adding click events as sources
```

Comment:
After trying this out I'm not as worried, to be honest. Passing null is
probably the uncommon case in real life anyway.

On Sun, Jun 19, 2016 at 11:03 PM, Filipp Riabchun <notifications@github.com>
wrote:

> In README.md
> <https://github.com/cyclejs/collection/pull/14#discussion_r67617530>:
>
> >  function TodoList (sources) {
> > -  const todoListItems = Collection(TodoListItem, sources);
> > -
> > -  const addTodoClick$ = sources.DOM.select('.add-todo').events('click');
> > +  const addTodo$ = sources.DOM
> > +    .select('.add-todo')
> > +    .events('click')
> > +    .mapTo(null); // to prevent adding click events as sources
>
> Well, usually you pass some payload like props anyway. But maybe we can
> eventuially figure out something more elegant, yes
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/cyclejs/collection/pull/14/files/946c9ff60eefcd9b253d0481c924675edb6a782d#r67617530>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAYUHRLpI-n-f6zG8O013ya9hrOOnqhBks5qNUwugaJpZM4I38NS>
> .
>